frontend/src/components/core/Editor.tsx

```diff
@@ -0,0 +1,24 @@
+import React, { ComponentProps } from 'react';
+import MonacoEditor from 'react-monaco-editor';
+
+type EditorProps = ComponentProps<typeof MonacoEditor>
+
+function ResizableMonacoEditor(props: EditorProps) {
+  const handleEditorDidMount = (editor: any) => {
+    window.addEventListener('resize', () => {
+      editor.layout();
+    });
+    window.addEventListener('secondaryPanelSizeChange', () => {
+      editor.layout();
+    });
+  };
+
+  return (
+    <MonacoEditor
+      {...props}
+      editorDidMount={handleEditorDidMount}
+    />
+  );
+}
```

Comment:
I will note that this might be tough given that editorDidMount seems to be specific to the way MonacoEditor implemented their API, however, maybe there's a way to use useEffect and have the dependency array include something like `[monacoEditorVariable]`? I'm not too sure if that's possible though

Reply:
That's fair -- if there's no workaround, this definitely satisfies the case for an exception as we're using an outside package for the coding editor. 👍

Reply:
What Alan said is correct -- there is an `EditorDidMount` property on this Monaco editor so I named the function correspondingly. `handleEditorDidMount()` is not associated with classes.